### ISSUE TITLE
Axe Dev Tools Accesibility Error

### ISSUE THREAD

**Issue by bsylwek** [joomla/joomla-cms#45679]
### Steps to reproduce the issue
I installed Axe webtools for Firefox and found errors. Has anyone solved the following Issues? They occur on all Joomla sites if accessibility is enabled and this is very important. It would be good if Joomla developers would look at this and fix the bugs in the next Joomla 4 patch release. This is very important right now. I have Joomla 4.4.13 and Cassiopeia theme.

[https://addons.mozilla.org/pl/firefox/a ... -devtools/](https://addons.mozilla.org/pl/firefox/addon/axe-devtools/)

Ensure elements that have scrollable content are accessible by keyboard
more information [https://dequeuniversity.com/rules/axe/4 ... AxeFirefox](https://dequeuniversity.com/rules/axe/4.10/scrollable-region-focusable?application=AxeFirefox)
Element Location:

.before-collapse

<ul class="before-collapse _access-scrollbar">

To solve this problem, you need to fix at least (1) of the following:

Element should have focusable content
Element should be focusable

Ensure the order of headings is semantically correct
more information

Element Location:

#mod-1

<h3 class="card-header " id="mod-1">Menu</h3>

To solve this problem, you need to fix the following: [https://dequeuniversity.com/rules/axe/4 ... AxeFirefox](https://dequeuniversity.com/rules/axe/4.10/heading-order?application=AxeFirefox)

Heading order invalid

Ensure all page content is contained by landmarks

Element Location:

.main-bottom > .card-header

<h3 class="card-header ">Subscribe</h3>

To solve this problem, you need to fix the following: [https://dequeuniversity.com/rules/axe/4 ... AxeFirefox](https://dequeuniversity.com/rules/axe/4.10/region?application=AxeFirefox)

Some page content is not contained by landmarks

Ensure all page content is contained by landmarks
[https://dequeuniversity.com/rules/axe/4 ... AxeFirefox](https://dequeuniversity.com/rules/axe/4.10/region?application=AxeFirefox)
Element Location:

.container-bottom-a

<div class="grid-child container-bottom-a">

To solve this problem, you need to fix the following:

Some page content is not contained by landmarks
Top

### Device and system info

* What version of Joomla were you using?
Joomla 4.4.13 and Cassiopeia theme

* What device were you using? (eg, phone, laptop)

* What web browser were you using? (eg, Chrome, Safari)

* If you were using assistive technology, please let us know the device or software you were using (eg VoiceOver, JAWS, sip-and-puff device, etc)

### Additional comments

If you know which [WCAG](https://www.w3.org/TR/WCAG/) or [ATAG](https://www.w3.org/TR/ATAG20/) criteria this violates, please indicate it here:

**Comment by wwwiiimmm** [comment]
When Joomla is installed _all_ modules have the tag div and the header is an h3.

The landmark issue can be solved by setting a different module tag.
Open the module and go to tab **Advanced**.
At the option **Module Tag** change _div_ to something else, the usual options may be _nav_, _aside_, _section_, but you have more options. It depends on what you are displaying and where...

Under the same tab, at option **Header Tag** you can set the tag for the module title.
Depending on the order (of modules) and/or selected tag, changing this may solve the invalid heading order.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45679">issues.joomla.org/tracker/joomla-cms/45679</a>.</sub>

**Comment by bsylwek** [comment]
For what purpose is it distributed with the Joomla system and other similar plugins with other competing CMS systems? I know that in reality usually disabled people have different other readers depending on the disability. This can often be confusing to the average person.

**Comment by bsylwek** [comment]
I understand, I just use tools Firefox Axe and https://wave.webaim.org that help in working on eliminating accessibility errors. Yes, I have the TAB key enabled to navigate the page and I have the menu enabled. Skip to navigation plugin
https://forum.joomla.org/viewtopic.php?t=994294
The text was listened to audio when the accessibility module was enabled. I currently do not have an audio reader because this add-on is disabled.

**Comment by bsylwek** [comment]
I think the issues was solved according to the instructions
Add missing role="button" to index.php Cassiopeia theme line
        <a href="#top" id="back-top" class="back-to-top-link" role="button" aria-label="<?php echo Text::_('TPL_CASSIOPEIA_BACKTOTOP'); ?>">
            <span class="icon-arrow-up icon-fw" aria-hidden="true"></span>
        </a>

**Comment by bsylwek** [comment]
Where? I can't quite find it. I haven't done this before ;-)

**Comment by bsylwek** [comment]
Where exactly? I can't find it. Am I having some kind of eclipse?

**Comment by drmenzelit** [comment]
Here is the explanation: https://manual.joomla.org/docs/get-started/git/github-joomla-cms/

**Comment by chmst** [comment]
Please read https://github.com/joomla/joomla-cms/issues/47215

**Comment by saniya-dogra** [comment]
Hi, I’m Saniya, a beginner contributor interested in GSoC 2026.
I would like to work on this issue. Could you guide me on how to start?

**Comment by brianteeman** [comment]
@saniya-dogra https://github.com/joomla/joomla-cms/issues/47215

**Comment by alikon** [comment]
> https://github.com/joomla/joomla-cms/issues/45679#issuecomment-4096778355
